src/pretalx/person/tasks.py

```diff
@@ -0,0 +1,60 @@
+import logging
+
+from django.core.files import File
+from django.core.files.temp import NamedTemporaryFile
+from django.dispatch import receiver
+from requests import get
+
+from pretalx.celery_app import app
+from pretalx.common.signals import periodic_task
+from pretalx.person.models.user import User
+
+logger = logging.getLogger(__name__)
+
+
+@app.task()
+def gravatar_cache(person_id: int):
+    user = User.objects.filter(pk=person_id, get_gravatar=True).first()
```

Comment:
Seems like this filter does not work correctly for the task that's being generated during the profile save step.

I'm always getting `gravatar_cache() was called for user 856, but user was not found or user has gravatar disabled`, but it gets fetched just fine using the periodic task. Not sure what's wrong here though.

Reply:
hmm, might be generated in a transaction, and the celery task is called too fast for the transaction to wrap up? I moved the whole logic to the model in 8669db20, as that's more future-safe for when we introduce a writable API. Can you check if that works?

Reply:
hm, now it doesn't update the avatar at all, even in the periodic task, even though the database field is `true`? I'm really confused.

Reply:
Grr, that was too early in the morning. Yeah, that does break. 58547479 fixes.